astroid/__pkginfo__.py

```diff
@@ -15,7 +21,12 @@
 
 extras_require = {}
 install_requires = ['lazy_object_proxy', 'six', 'wrapt']
-extras_require[':python_version<"3.4"'] = ['enum34', 'singledispatch']
+
+if py_version < (3, 4) and setuptools_version < '21.0.0':
```

Comment:
You can't just compare strings:

```
>>> '3.0.0' < '21.0.0'
False
```

See what [pytest does](https://github.com/pytest-dev/pytest/blob/master/setup.py#L44) here.